0000-catalog.md

```diff
@@ -0,0 +1,427 @@
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="7d1734adfee4927"
+        ),
+        PropertyValue(
+            name="status",
+            data_type=PropertyDefinition.DataType.ENUM,
+            enum_value=1, # ACTIVE (default)
+        ),
+        PropertyValue(
+            name="price"
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="1-10:$50, 10-50:$40, 50+:$30"  # price windows
+        ),
+    ])
+```
+
+## Transaction Payload and Execution (catalog operations)
+
+CatalogPayload Transaction
+CatalogPayload contains an action enum and the associated action payload. This allows for the action payload to be dispatched to the appropriate logic.
+
+Only the defined actions are available and only one action payload should be defined in the CatalogPayload.
+
+message CatalogPayload { 
+    enum Actions { 
+        UNSET_ACTION = 0; 
+        CATALOG_CREATE = 1; 
+        CATALOG_DELETE = 2; 
+        CATALOG_REPLICATE = 3; 
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    CatalogCreateAction catalog_create = 3; 
+    CatalogDeleteAction catalog_delete = 4; 
+    CatalogReplicateAction catalog_replicate = 5; 
+} 
+
+### CatalogCreateAction
+CatalogCreateAction adds a new catalog to state. The transaction should be submitted by an agent, which is identified by its signing key, acting on behalf of the organization that corresponds to the owner in the create transaction. (Organizations and agents are defined by the Pike smart contract.)
+
+message CatalogCreateAction { 
+    // GS1 Company Prefix from owner + catalog_id are use as 
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    string expiry_date = 4;
+    repeated PropertyValues properties = 4; 
+} 
+
+Validation requirements:
+
+- If a catalog with catalog_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_catalog for the organization, otherwise the transaction is invalid.
+- If the product_namespace is GS1, the organization must contain a GS1 Company Prefix in its metadata (gs1_company_prefixes), and the prefix must match the company prefix in the product_id, which is a GTIN if GS1, otherwise the transaction is invalid.
+- The properties must be valid for the product_namespace. For example, if the product is GS1 product, its properties must only contain properties that are included in the GS1 Schema. If it includes a property not in the GS1 Schema the transaction is invalid.
+
+AddProductsToCatalog
+RemoveProductsFromCatalog
+ActivateProduct
+DeactivateProduct
+DiscontinueProduct
+
+# Rationale and alternatives
```

Comment:
Add a note about why we are using a reference-based implementation as opposed to a list concept